src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java

```diff
@@ -35,45 +33,51 @@ URI getRequestUrl(SteamWebApiRequest request) throws SteamApiException {
 		String scheme = getProtocol();
 		String host = request.getBaseUrl();
 		String path = getRequestPath(request);
-		List<NameValuePair> parameters = getRequestParameters(request.getParameters());
+		String query = getRequestQuery(request.getParameters());
 
-		URI requestUrl = getRequestUri(scheme, host, path, parameters);
-
-		return requestUrl;
+		return getRequestUri(scheme, host, path, query);
 	}
 
-	String getRequestPath(SteamWebApiRequest request) {
-		StringBuilder requestPath = new StringBuilder();
-
-		requestPath.append("/");
-		requestPath.append(request.getApiInterface().toString());
-		requestPath.append("/");
-		requestPath.append(request.getInterfaceMethod().toString());
-		requestPath.append("/");
-		requestPath.append(request.getVersion().toString());
-
-		return requestPath.toString();
+	String getRequestPath(SteamWebApiRequest request) {
+		// creates "/ApiInterface/InterfaceMethod/Version"
+		return Stream.of(
+				request.getApiInterface(),
+				request.getInterfaceMethod(),
+				request.getVersion()
+		)
+		.map(Object::toString)
+		.collect(joining("/", "/", ""));
 	}
 
-	List<NameValuePair> getRequestParameters(Map<String, String> parametersMap) {
-
-		List<NameValuePair> nvps = new ArrayList<>();
+	String getRequestQuery(Map<String, String> parameters) throws SteamApiException {
+		if (getKey() == null)
+			throw new SteamApiKeyException("Steam API key is not present or null");
+
+		for (var e : parameters.entrySet())
+			if (e.getKey() == null)
+				throw new SteamApiException("The key of the parameter with the value '" + e.getValue() + "' is null");
+
+		parameters.put("key", getKey());
+		return parameters.entrySet().stream()
+				.map(e -> {
+					String v = e.getValue(); // if the value of the parameter is null, we replace it with an empty String
+					return e.getKey() + '=' + URLEncoder.encode(v == null ? "" : v, UTF_8);
+				})
+				.collect(joining("&"));
+	}
 
-		nvps.add(new BasicNameValuePair("key", getKey()));
+	URI getRequestUri(String scheme, String host, String path, String query) throws SteamApiException {
+		var uri = new StringBuilder();
 
-		parametersMap.entrySet().forEach(param -> nvps.add(new BasicNameValuePair(param.getKey(), param.getValue())));
+		uri.append(scheme);
+		uri.append("://");
+		uri.append(host);
+		uri.append(path);
+		uri.append("?");
+		uri.append(query);
 
-		return nvps;
-	}
-
-	URI getRequestUri(String scheme, String host, String path, List<NameValuePair> parameters) throws SteamApiException {
 		try {
-			return new URIBuilder()
-					.setScheme(scheme)
-					.setHost(host)
-					.setPath(path)
-					.setParameters(parameters)
-					.build();
+			return new URI(uri.toString());
 		} catch (URISyntaxException e) {

```

Comment:
Honestly not sure if it was even covered before, but in my IDE coverage it says that this `catch` case is not covered by the unit test. Maybe you could add one for completeness sake?